extension/intents/navigation/navigation.js

```diff
@@ -9,32 +9,46 @@ import * as browserUtil from "../../browserUtil.js";
 const QUERY_DATABASE_EXPIRATION = 1000 * 60 * 60 * 24 * 30; // 30 days
 const queryDatabase = new Map();
 
+function database(query, tab, url) {
```

Comment:
This name isn't very descriptive, maybe it could be `saveTabQueryToDatabase(...)` ?

Reply:
Even I like this name too I will update this